e2e/src/componentsController.spec.ts

```diff
@@ -0,0 +1,639 @@
+      await componentsController.createMeteredComponent(productFamilyId, {
+        ...componentPayload,
+      });
+
+    return createComponentReponse.result;
+  };
+
+  describe('Find Component', () => {
+    test('should find component with valid id', async () => {
+      const newProductPayload: CreateOrUpdateProductRequest = {
+        product: {
+          name: `product-read-component-name-${uid()}`,
+          handle: `product-read-component-${uid()}`,
+          description: '',
+          requireCredit_card: true,
+          priceInCents: BigInt(1000),
+          interval: 1,
+          intervalUnit: IntervalUnit.Month,
+        },
+      };
+
+      const productFamilyId = productFamily?.id || 0;
+
+      const meteredComponent: CreateMeteredComponent = {
+        meteredComponent: {
+          productFamilyId,
+          name: `component-name-valid${uid()}`,
+          description: 'test',
+          quantity: 1,
+          unitName: 'test',
+          price: 1,
+          handle: uid(),
+          pricingScheme: PricingScheme.PerUnit,
+          unitPrice: 1,
+        },
+      };
+
+      const { component } = await prepareDataForComponentResponse(
+        newProductPayload,
+        productFamilyId,
+        meteredComponent
+      );
+
+      const componentHandle = component.handle || '';
+
+      const componentsControllerResponse =
+        await componentsController.findComponent(componentHandle);
+
+      expect(componentsControllerResponse.statusCode).toBe(200);
+      expect(componentsControllerResponse.result.component?.handle).toBe(
+        componentHandle
+      );
+    });
+
+    test('should not find component with invllid id', async () => {
+      const newProductPayload = {
+        product: {
+          name: `product-find-component-name-invalid-id-${uid()}`,
+          handle: `product-find-component-${uid()}`,
+          description: '',
+          requireCredit_card: true,
+          priceInCents: BigInt(1000),
+          interval: 1,
+          intervalUnit: IntervalUnit.Month,
+        },
+      };
+      const productFamilyId = productFamily?.id || 0;
+
+      const meteredComponent: CreateMeteredComponent = {
+        meteredComponent: {
+          productFamilyId,
+          name: `component-name-valid${uid()}`,
+          description: 'test',
+          quantity: 1,
+          unitName: 'test',
+          price: 1,
+          handle: uid(),
+          pricingScheme: PricingScheme.PerUnit,
+          unitPrice: 1,
+        },
+      };
+
+      await prepareDataForComponentResponse(
+        newProductPayload,
+        productFamilyId,
+        meteredComponent
+      );
+
+      const componentsControllerInavlidResponse =
+        componentsController.findComponent(invalidComponentId);
+
+      expect(componentsControllerInavlidResponse).rejects.toThrow();
+
+      await componentsControllerInavlidResponse.catch((reason) => {
+        expect(reason.statusCode).toBe(404);
```

Comment:
Same here, we should check error messages to ensure customer receives errors properly.